protocol/src/common/proposed.fileSystemProvider.ts

```diff
@@ -0,0 +1,621 @@
+	/**
+     * If a folder should be recursively subscribed to
+     */
+	recursive: boolean;
+
+	/**
+     * Folders or files to exclude from being watched.
+     */
+	excludes: string[];
+}
+
+// StopWatching Notification (client->server) fileSystem/watch
+
+/**
+ * The fileSystem stop watching notification definition. Stop watching notifications are sent from client to server to
+ * unsubscribe from a watched file or folder.
+ *
+ * @since 3.17.0 - proposed state
+ */
+export namespace StopWatchingNotification {
+	export const type = new ProtocolNotificationType<StopWatchingParams, void>('fileSystem/stopWatching');
+	export type HandlerSignature = NotificationHandler<StopWatchingParams>;
+	export type MiddlewareSignature = (params: StopWatchingParams, next: HandlerSignature) => void;
+}
+
+/**
+ * Parameters of the stop watching notification.
+ *
+ * @since 3.17.0 - proposed state
+ */
+export interface StopWatchingParams {
+	/**
+     * The subscription id.
+     */
+	subscriptionId: string;
+}
+
+// Stat Request (client->server) fileSystem/stat
+
+/**
+ * The fileSystem stat rquest definition. Stat requests are sent from the client to the server to request metadata about a URI.
+ *
+ * @since 3.17.0 - proposed state
+ */
+export namespace StatRequest {
+	export const method: 'fileSystem/stat' = 'fileSystem/stat';
+	export const type = new ProtocolRequestType<FileStatParams, FileStatResponse, void, FileSystemErrorType, FileSystemProviderRegistrationOptions>(method);
+	export type HandlerSignature = RequestHandler<FileStatParams, FileStatResponse, FileSystemErrorType>;
+}
+
+/**
+ * Parameters of the file stat request.
+ *
+ * @since 3.17.0 - proposed state
+ */
+export interface FileStatParams {
+	/**
+     * The uri to retrieve metadata about.
+     */
+	uri: string;
+}
+
+/**
+ * Response for the file stat request.
+ *
+ * @since 3.17.0 - proposed state
+ */
+export interface FileStatResponse {
+	/**
+     * The type of the file, e.g. is a regular file, a directory, or symbolic link
+     * to a file/directory.
+     *
+     * *Note:* This value might be a bitmask, e.g. `FileType.File | FileType.SymbolicLink`.
+     */
+	type: FileType;
+
+	/**
+     * The creation timestamp in milliseconds elapsed since January 1, 1970 00:00:00 UTC.
+     */
+	ctime: number;
+
+	/**
+     * The modification timestamp in milliseconds elapsed since January 1, 1970 00:00:00 UTC.
+     *
+     * *Note:* If the file changed, it is important to provide an updated `mtime` that advanced
+     * from the previous value. Otherwise there may be optimizations in place that will not show
+     * the updated file contents in an editor for example.
+     */
+	mtime: number;
+
+	/**
+     * The size in bytes.
+     *
+     * *Note:* If the file changed, it is important to provide an updated `size`. Otherwise there
+     * may be optimizations in place that will not show the updated file contents in an editor for
+     * example.
+     */
+	size: number;
```

Comment:
Perhaps `unumber` can be used here instead to indicate that these must be non-negative numbers.